### PR TITLE
Refine logging in pkg/dashboard/apiserver/auth/gcp

### LINKED ISSUE #2873: Refine logging in `pkg/dashboard/apiserver/auth/gcp`, move package level variable `log` into struct `Service` as a field
> Please leave comments with your idea if you are interested in this issue. And comment `/assign` to operate self-assign before you work on it.

### PR DESCRIPTION
**Pull request by yoonian**
### What problem does this PR solve?

Close #2873 

### What's changed and how it works?

Moving package level variable log into struct Service as a field

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.3
  - [ ] release-2.2
  - [ ] release-2.1

### Checklist

CHANGELOG

- [x] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### CODE REVIEW

File: pkg/dashboard/apiserver/auth/gcp/fx.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"github.com/go-logr/logr"
+
+	config "github.com/chaos-mesh/chaos-mesh/pkg/config/dashboard"
+)
+
+func Bootstrap(conf *config.ChaosDashboardConfig, logger logr.Logger) *Service {
+	service, _ := NewService(conf, logger.WithName("gcp auth api"))
+	return service
+}
```
Comment (STRRL) [inline pkg/dashboard/apiserver/auth/gcp/fx.go:27]: please do not ignore that error returned by `NewService`. After that, `Bootstrap` do not do extra things other than `NewService`, please use `NewService` with `fx` directly.

File: pkg/dashboard/apiserver/auth/gcp/service.go [hunk 2]
```diff
@@ -21,28 +21,29 @@ import (
...
 }
 
 // NewService returns an experiment service instance.
 func NewService(
 	conf *config.ChaosDashboardConfig,
+	logger logr.Logger,
 ) (*Service, error) {
 	rootUrl, err := url.Parse(conf.RootUrl)
 	if err != nil {
+		logger.Error(err, "fail to parse rootUrl")
```
Comment (STRRL) [inline pkg/dashboard/apiserver/auth/gcp/service.go:46]: as we have returned the `error`, do not log it again.

**Review by STRRL — COMMENTED** [review]
rest LGTM

File: pkg/dashboard/apiserver/auth/gcp/fx.go (hunk 1 above)
Comment (yoonian) [reply pkg/dashboard/apiserver/auth/gcp/fx.go:27]: Thanks, I'll apply.

File: pkg/dashboard/apiserver/auth/gcp/service.go (hunk 2 above)
Comment (yoonian) [reply pkg/dashboard/apiserver/auth/gcp/service.go:46]: Okay!

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Review by xlgao-zju — APPROVED** [review]
LGTM

**Pull request merged**